netty/src/test/java/io/grpc/netty/TlsTest.java

```diff
@@ -185,6 +187,52 @@ public void basicClientServerIntegrationTest() throws Exception {
     client.unaryRpc(SimpleRequest.getDefaultInstance());
   }
 
+  /**
+   * Test that a client fails to send requests over TLS to a server with an unexpected hostname listed
+   * in an otherwise valid certificate.
+   */
+  @Test
+  public void serverRejectedBadHostname() throws Exception {
+    // Create & start a server.
+    File serverCertFile = TestUtils.loadCert("server1.pem");
+    File serverPrivateKeyFile = TestUtils.loadCert("server1.key");
+    X509Certificate[] serverTrustedCaCerts = {
+            TestUtils.loadX509Cert("ca.pem")
+    };
+    server = serverBuilder(0, serverCertFile, serverPrivateKeyFile, serverTrustedCaCerts)
+            .addService(new SimpleServiceImpl())
+            .build()
+            .start();
+
+    // Create a client.
+    File clientCertChainFile = TestUtils.loadCert("client.pem");
+    File clientPrivateKeyFile = TestUtils.loadCert("client.key");
+    X509Certificate[] clientTrustedCaCerts = {
+            TestUtils.loadX509Cert("ca.pem")
+    };
+
+    // Override authority on the client to trigger a hostname verification failure
+    channel = NettyChannelBuilder.forAddress("localhost", server.getPort())
+            .overrideAuthority("i.am.a.bad.hostname")
+            .negotiationType(NegotiationType.TLS)
+            .sslContext(clientContextBuilder
+              .keyManager(clientCertChainFile, clientPrivateKeyFile)
+              .trustManager(clientTrustedCaCerts)
+              .build()
+            )
+            .build();
+    SimpleServiceGrpc.SimpleServiceBlockingStub client = SimpleServiceGrpc.newBlockingStub(channel);
+
+    // Send an actual request, via the full GRPC & network stack, and check that a proper
+    // response comes back.
+    try {
+      client.unaryRpc(SimpleRequest.getDefaultInstance());
```

Comment:
A `fail()` is needed in the try after `unaryRpc()`, otherwise if the RPC succeeds the test will still pass. (This is why the CI failed, because there's a checker for mistakes like that.)